src/main/java/org/xbill/DNS/Serial.java

```diff
@@ -34,17 +33,14 @@
 	if (serial2 < 0 || serial2 > MAX32)
 		throw new IllegalArgumentException(serial2 + " out of range");
 	long diff = serial1 - serial2;
-	if (diff >= MAX32)
```

Comment:
This if-elseif clause appeared to be redundant, but please sanity check me.  The long->int conversion seems to provide the exact same values as we currently check for in the SerialTest test_compare_boundary
https://github.com/dnsjava/dnsjava/blob/6d50d8cc03e412642f28a0890c626f1283b8a86e/src/test/java/org/xbill/DNS/SerialTest.java#L129

Reply:
test_compare_boundary IMO asserts wrong values, arg2 is definitely not larger than arg1:

	long arg1 = 0xFFFFFFFFL;
	long arg2 = 0;
	int ret = Serial.compare( arg1, arg2 );
	assertEquals( -1, ret );

In master, we can simply replace the entire block with `return Long.compare(serial1, serial2);`, which reduces the method to sanity checking the inputs (which in turn is only necessary because of Java's horrific stupidity not having unsigned integers).
If this should be fixed in 2.1.x too, just compare the longs instead of calculating: `return s1 < s2? - 1 : (s1 == s2 ? 0 : 1);`